src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/InternalCaseToLegallyRepresentedHandler.java

```diff
@@ -0,0 +1,163 @@
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.LegRepAddressUk;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.DispatchPriority;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.AddressUk;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ref.OrganisationEntityResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.clients.ProfessionalOrganisationRetriever;
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.clients.model.ccd.Organisation;
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.clients.model.ccd.OrganisationPolicy;
+
+@Slf4j
+@Component
+public class InternalCaseToLegallyRepresentedHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final ProfessionalOrganisationRetriever professionalOrganisationRetriever;
+
+    public InternalCaseToLegallyRepresentedHandler(
+        ProfessionalOrganisationRetriever professionalOrganisationRetriever
+    ) {
+        this.professionalOrganisationRetriever = professionalOrganisationRetriever;
+    }
+
+    @Override
+    public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                && callback.getEvent() == Event.NOC_REQUEST
+                && isInternalCase(callback.getCaseDetails().getCaseData());
+    }
+
+    public DispatchPriority getDispatchPriority() {
+        return DispatchPriority.EARLIEST;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callbackStage,
+                                                        Callback<AsylumCase> callback) {
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        //Clearing other unneeded Internal case data
+        asylumCase.clear(APPEAL_SUBMISSION_INTERNAL_DATE);
+        asylumCase.clear(TRIBUNAL_RECEIVED_DATE);
+        asylumCase.clear(INTERNAL_APPELLANT_EMAIL);
+        asylumCase.clear(INTERNAL_APPELLANT_MOBILE_NUMBER);
+
```

Comment:
Do we not need to clear the isAdmin flag? Approved PR in case we still need isAdmin flag, and if we don't I guess you can just clear it like you have for the others.

Reply:
I just set the isAdmin flag to No at the bottom of that handler after the legal rep details being added. Looking back I probably should have had it at the top though beside these clears.